### PR TITLE
"General IT" section, "Backend" section

### PR DESCRIPTION
**Pull request by kirilltitov**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,18 +27,21 @@
...
 * Data structures and algorithms
 * Design patterns
+* Code smells
+* Refactoring techniques
 * SOLID, KISS, YAGNI
+* TDD, BDD
+* Functional programming
 * Licenses
 * Semantic versioning
-* Character encodings
```
Comment (kirilltitov) [inline README.md]: Probably need more than one item :D /shrug

File: README.md [hunk 2]
```diff
@@ -50,6 +57,20 @@ How modern networks work from developers POV
...
+
+// TODO
+
+# Frontend (General)
+
+## Frontend Prerequisites
+
+Basic development principles, general for any development path
+
+* Character encodings
```
Comment (kirilltitov) [inline README.md]: Probably need more than one item :D /shrug

File: README.md (hunk 1 above)
Comment (apetushkov) [inline README.md]: We need it for both Frontend, Backend and Mobile platforms. Also makes sense to make it "Character encodings, collation"

File: README.md (hunk 2 above)
Comment (apetushkov) [reply README.md]: Not a frontend-specific knowledge

Comment (kirilltitov) [reply README.md]: Well, this whole section becomes obsolete then

Comment (apetushkov) [reply README.md]: Yep

**Pull request closed without merging**